### PR TITLE
Support trust domain

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Fix https://github.com/istio/ztunnel/issues/435

### CODE REVIEW

File: src/workload.rs [hunk 1]
```diff
@@ -94,6 +94,8 @@ pub struct Workload {
     #[serde(default)]
     pub namespace: String,
     #[serde(default)]
+    pub trust_domain: String,
```
Comment (zhlsunshine) [inline src/workload.rs:97]: may need to display `trust_domain` in `fmt::Display for Workload` if add this field.

**Comment by zhlsunshine** [comment]
IMO, I think that the `trust_domain` information should be added in `ProxyConfig` not in every `Workload`.

**Comment by zhlsunshine** [comment]
IMO, I think that the `trust_domain` information should be added in or be fetched from `ProxyConfig`, not in every `Workload`.

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
> IMO, I think that the `trust_domain` information should be added in or be fetched from `ProxyConfig`, not in every `Workload`.

Its suppose to be set if its *different* from the default trust domain (multi-trust domain meshes)

**Pull request merged**